### ISSUE TITLE
Upgrade to Rider 2023.1

### ISSUE THREAD

**Issue by asidorowicz** [cake-build/cake-rider#325]
Love the plugin, however it does not work with the latest version of Rider
I tried editing the jar to force it to load, and it does. The settings menu hangs however, so it might require a little work.
The tasks do run though, and the rest seems to work well

**Comment by sebastienpa** [comment]
+1

**Comment by baynezy** [comment]
Any update on how this is going?

**Comment by evhenii-komar** [comment]
Still waiting on it.

**Comment by LoveDuckie** [comment]
I wish it was possible to +1 react to a comment, so I apologise for how asinine this comment is! 😄 

I also love the plugin and waiting for the 2023 support. Going to roll my sleeves up and take a look and what's required for the fix and if there's any breaking API change.

**Comment by nils-a** [comment]
@LoveDuckie there are some minor breaking changes, nothing serious though.
The main obstacle currently is the settings pages crashing when the settings are opening.
So far I have somehow been unable to get a handle on that.

**Comment by nils-a** [comment]
@LoveDuckie I think I found the problem - something in the way the icons were loaded in the settings pages.

You wouldn't have the cycles to check out #351 and take it for a spin, would you? 😉

**Comment by nils-a** [comment]
I have made a preview version available in the [beta chanel](https://plugins.jetbrains.com/plugin/15729-cake-for-rider/versions/beta/383047) for everyone to test.

**Comment by LoveDuckie** [comment]
Hey @nils-a ! Thanks for pushing that out. The version appears to be working on the 2023 version of Rider that I'm using.

I just wanted to say that I've noticed that the same task will appear multiple times (one per `.cs` file) in the "Tasks" panel for the extension, _**if**_ the derived Cake FrostingTask type is partially defined (i.e. `partial`) in multiple `.cs` files. I can open a new ticket for this if you like. Not sure if this is existing behaviour before the 2023 fix.

**Comment by nils-a** [comment]
@LoveDuckie yes, I'd appreciate it if you could open a new issue.
(That way I won't forget about it 😬)
